animation_nodes/utils/attributes.py

```diff
@@ -20,19 +20,24 @@ def getAttributeGetter(propName):
 def getMultiAttibuteSetter(propNames):
     code = "def setter(owner, values):\n"
     for i, prop in enumerate(propNames):
-        line = getAttributeSetterLine("owner", prop, "values[{}]".format(i))
-        code += "    " + line + "\n"
+        lines = getAttributeSetterLines("owner", prop, "values[{}]".format(i))
+        code += ''.join("    " + line + "\n" for line in lines)
     code += "    pass"
     variables = {}
     exec(code, variables)
     return variables["setter"]
 
-def getAttributeSetterLine(objectName, propName, valueName):
-    if propName.startswith('["'):
+def getAttributeSetterLines(objectName, propName, valueName):
+    if propName.startswith('['):
         # Path is a named attribute
-        return '{}{} = {}'.format(objectName, propName, valueName)
+        path = "{}{}".format(objectName, propName)
     else:
-        return '{}.{} = {}'.format(objectName, propName, valueName)
+        path = "{}.{}".format(objectName, propName)
+
+    return (
```

Comment:
Use `yield` instead of return. Double quotes instead of singles quotes. Comments go above statements. Use f-strings.

```python
# Try to cast to existing property type.
yield f"try: {path} = type({path})({valuename})"
# Property does not exist, default to new float.
yield f"except: {path} = {valueName}"
```